### PR TITLE
Fix restart flow for preview-dart-2 mode.

### PR DESCRIPTION
**Pull request by aam**
Restart in preview-dart-2 needs to use kernel file and it has to be complete, rather than incremental kernel file.

### CODE REVIEW

**Review by scheglov — APPROVED** [review]

File: packages/flutter_tools/lib/src/run_hot.dart [hunk 1]
```diff
@@ -334,6 +334,12 @@ class HotRunner extends ResidentRunner {
     }
 
     final Stopwatch restartTimer = new Stopwatch()..start();
+    if (previewDart2)
```
Comment (a-siva) [inline packages/flutter_tools/lib/src/run_hot.dart]: Not sure about the style guide but feel the code will be more readable in this case if you added '{' '}' for this if statement.

**Review by a-siva — COMMENTED** [review]
LGTM

**Review by a-siva — APPROVED** [review]

File: packages/flutter_tools/lib/src/run_hot.dart (hunk 1 above)
Comment (aam) [reply packages/flutter_tools/lib/src/run_hot.dart]: You are right, https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-braces-for-one-line-long-control-structure-statements talks about avoiding curly braces only around one-line long control structure statements.

**Comment by aam** [comment]
PTAL since I added logic to do batch compilation on restart as a workaround for https://github.com/dart-lang/sdk/issues/31043

**Review by a-siva — APPROVED** [review]

**Comment by aam** [comment]
I need to roll the engine as part of this PR to make sure that preview-dart-2 tests still pass.

@jason-simmons

**Review by jason-simmons — APPROVED** [review]

**Comment by cbracken** [comment]
This appears to have caused a massive regression in the following hot_mode_dev_cycle__preview_dart_2_benchmark benchmarks:
  * hotReloadDevFSSyncMilliseconds
  * hotReloadFlutterReassembleMilliseconds
  * hotReloadMillisecondsToFrame
  * hotReloadVMReloadMilliseconds
  * hotRestartMillisecondsToFrame

**Pull request merged**